docs/howto/app.rst

```diff
@@ -16,3 +16,92 @@ In the first two cases, it's important to specify the location to your tasks usi
         connector=connector,
         import_paths=["dotted.path.to", "all.the.modules", "that.define.tasks"]
     )
+
+Initiate and terminate the connection to the database
+-----------------------------------------------------
+
+The app must be opened at the beginning of the program to create the connection pool to
+the database, and eventually closed to properly terminate it.
+
+The app implements several ways to do this.
+
+Explicitly open and close the app
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The app can be opened using the ``open`` method (for sync case) or ``open_async``
+method (for async case). It eventually needs to be closed, using the ``close`` method
+(for sync case) or ``close_async`` (for async case).
+
+For sync case::
+
+    app.open()
+    ...
+    app.close()
+
+For async case::
+
+    await app.open_async()
+    ...
+    await app.close_async()
+
+
+Open the app and automatically close it with a context manager
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The app can be opened with a context manager, which will automatically close it when
+leaving the context.
+
+For sync case::
+
+    with app.open():
+        pass
+
+For async case::
+
+    async with app.open_async():
+        pass
+
+
+Create and open the app in a single line
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Both ``open`` and ``open_async`` return the current app when they are called or
```

Comment:
Strictly speaking `open_async` returns an awaitable, which `__await__` method returns the application. Do we want to be more accurate here in the documentation?

Reply:
You are right. I tried to keep this as simple as possible. Is the following better ?
```suggestion
Both ``open`` and ``open_async`` return an ``AwaitableContext`` object that returns the current app when it is entered, or awaited
```
It is more accurate, but I am afraid it improves complexity too much.

Reply:
Yep, I think it's counter-productive to mention the AwaitableContext here. I'm comfortable with the original sentence.